flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClient.java

```diff
@@ -243,46 +247,46 @@ public static JobExecutionResult awaitJobResult(JobListeningContext listeningCon
 		final JobID jobID = listeningContext.getJobID();
 		final ActorRef jobClientActor = listeningContext.getJobClientActor();
 		final Future<Object> jobSubmissionFuture = listeningContext.getJobResultFuture();
+		final FiniteDuration askTimeout = listeningContext.getTimeout();
 		// retrieves class loader if necessary
 		final ClassLoader classLoader = listeningContext.getClassLoader();
 
+		// wait for the future which holds the result to be ready
+		// ping the JobClientActor from time to time to check if it is still running
 		while (!jobSubmissionFuture.isCompleted()) {
 			try {
-				Thread.sleep(250);
-			} catch (InterruptedException e) {
-				throw new JobExecutionException(jobID, "Interrupted while waiting for execution result.", e);
-			}
-
-			try {
-				Await.result(
-					Patterns.ask(
-						jobClientActor,
-						JobClientMessages.getPing(),
-						Timeout.durationToTimeout(AkkaUtils.getDefaultTimeout())),
-					AkkaUtils.getDefaultTimeout());
+				Await.ready(jobSubmissionFuture, askTimeout);
 			} catch (Exception e) {
```

Comment:
Ah, I thought you were commenting on the inner catch block. Yes, it makes sense to catch only `TimeoutException` and `InterruptedException` here for `Await.ready`. `Await.result` actually throws Exception.